src/main/java/org/jabref/gui/StateManager.java

```diff
@@ -57,8 +61,40 @@
     private final EasyBinding<Double> tasksProgress = EasyBind.reduce(backgroundTasks, tasks -> tasks.filter(Task::isRunning).mapToDouble(Task::getProgress).average().orElse(1));
     private final ObservableMap<String, DialogWindowState> dialogWindowStates = FXCollections.observableHashMap();
 
+    private final ObservableList<SidePaneType> visiblePanes = FXCollections.observableArrayList();
+
+    private final BooleanProperty groupsPaneVisible = new SimpleBooleanProperty();
+    private final BooleanProperty openOfficePaneVisible = new SimpleBooleanProperty();
+    private final BooleanProperty webSearchPaneVisible = new SimpleBooleanProperty();
+
     public StateManager() {
         activeGroups.bind(Bindings.valueAt(selectedGroups, activeDatabase.orElse(null)));
+
+        visiblePanes.addListener((ListChangeListener<? super SidePaneType>) change -> {
+            while (change.next()) {
+                if (change.wasAdded()) {
+                    SidePaneType visibleComponent = change.getAddedSubList().get(0);
+                    sidePaneComponentVisiblePropertyFor(visibleComponent).set(true);
+                    Globals.prefs.getSidePanePreferences().visiblePanes().add(visibleComponent);
+                } else if (change.wasRemoved()) {
+                    SidePaneType hiddenComponent = change.getRemoved().get(0);
+                    sidePaneComponentVisiblePropertyFor(hiddenComponent).set(false);
+                    Globals.prefs.getSidePanePreferences().visiblePanes().remove(hiddenComponent);
+                }
+            }
+        });
+    }
+
+    public BooleanProperty sidePaneComponentVisiblePropertyFor(SidePaneType component) {
+        return switch (component) {
+            case WEB_SEARCH -> webSearchPaneVisible;
+            case GROUPS -> groupsPaneVisible;
+            case OPEN_OFFICE -> openOfficePaneVisible;
+        };
```

Comment:
I can't help, but my gut feeling is that this sidepane logic simply does not belong in the StateManager.

Reply:
I'll move the binding to `SidePaneViewModel`